src/nrniv/prcellstate.cpp

```diff
@@ -42,16 +39,16 @@ static void pr_memb(int type, Memb_list* ml, int* cellnodes, NrnThread& nt, FILE
     }
 }
 
-static void pr_netcon(NrnThread& nt, FILE* f) {
-    if (pntindex == 0) {
+static void pr_netcon(NrnThread& nt, FILE* f, const std::map<void*, int>& pnt2index) {
+    if (pnt2index.empty()) {
         return;
     }
     // pnt2index table has been filled
 
     // List of NetCon for each of the NET_RECEIVE point process instances
     // ... all NetCon list in the hoc NetCon cTemplate
-    NetConList** nclist = new NetConList*[pntindex];
-    for (int i = 0; i < pntindex; ++i) {
+    NetConList** nclist = new NetConList*[pnt2index.size()];
+    for (int i = 0; i < pnt2index.size(); ++i) {
```

Comment:
I just realized ```int i``` will generate a warning (with some compilers) and should be changed to ```size_t i```.
If you don't mind, maybe now is a good time to compile this manually with ```-Wall```. I do that by copy/pasting the relevant compile command output from ```make VERBOSE=1``` and adding that arg.  But no need to bother if it is too tedious for this PR.